thirdparty/github.go

```diff
@@ -873,66 +1197,153 @@ func AppAuthorizedForOrg(ctx context.Context, token, requiredOrganization, name
 }
 
 func GitHubUserPermissionLevel(ctx context.Context, token, owner, repo, username string) (string, error) {
-	httpClient := getGithubClientRetryWith404s(token, "GithubUserPermissionLevel")
-	defer utility.PutHTTPClient(httpClient)
+	installationToken, _ := GetInstallationToken(ctx, owner, repo, nil)
 
+	httpClient := getGithubClientRetryWith404s(installationToken, "GithubUserPermissionLevel")
+	defer utility.PutHTTPClient(httpClient)
 	client := github.NewClient(httpClient)
 
 	permissionLevel, _, err := client.Repositories.GetPermissionLevel(ctx, owner, repo, username)
+	// fallback to not using github app on error
+	if err == nil && permissionLevel != nil && permissionLevel.Permission != nil {
```

Comment:
same as above,  I think it might feel more readable if the fallback was wrapped in an err != nil instead of exiting if err==nil.

Reply:
discussed!